### PR TITLE
chore: upgrade docs tooling

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
This upgrades the docs tooling and attempts to clean up the documentation to include only what end-users of the module actually need to know about.

# Checklist

- [ ] All acceptance criteria are met.
- [ ] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [ ] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by ThisIsMissEm** [comment]
@NSeydoux I've added a usage example here: https://solid-client-notifications-js-git-chore-upgrade-d-84d5fc-inrupt.vercel.app/classes/websocketNotification.WebsocketNotification.html

Though it is quite minimal

File: docs/api/source/index.rst [hunk 1]
```diff
@@ -1,14 +1,103 @@
...
-==============================
+====================================================
+@inrupt/solid-client-notifications API Documentation
+====================================================
 
+`@inrupt/solid-client-notifications
+<https://npmjs.com/package/@inrupt/solid-client-notifications>`__ is a
+JavaScript library for requesting and managing access given to an agent for a
+resource. These access grants are represented by Verifiable Credentials, signed
+by an Issuer associated to the Pod server where the resources are hosted.
```
Comment (NSeydoux) [inline docs/api/source/index.rst:11]:
```suggestion
<https://npmjs.com/package/@inrupt/solid-client-notifications>`__ is a
JavaScript library for subscribing to notifications sent by a Solid server.
These notifications may be sent synchronously e.g., through WebSockets,
or asynchronously e.g. via WebHooks. ? Notifications are typically sent
when a resource is updated/deleted, or when a child resource is added
to/removed from a container.
```

**Review by NSeydoux — APPROVED** [review]

File: src/notification.ts [hunk 2]
```diff
@@ -18,36 +18,18 @@
...
+  BaseNotificationOptions,
+  NegotiationInfo,
+  NotificationConnectionInfo,
+} from "./interfaces";
+
+/**
+ * @internal
+ */
 export class BaseNotification {
   topic: string;
```
Comment (NSeydoux) [inline src/notification.ts:34]: All of the properties of this class should be private, and not appear in the documentation. All these values are provided by the constructor or during the negotiation with the notification endpoint, they should neither be set a posteriori not be read by a third party.

File: docs/api/source/index.rst (hunk 1 above)
Comment (ThisIsMissEm) [reply docs/api/source/index.rst]: Have updated it, but with a refined version of the text above.

File: e2e/browser/testApp/package.json [hunk 3]
```diff
@@ -10,17 +10,17 @@
   },
   "dependencies": {
     "@inrupt/solid-client-authn-browser": "^1.11.7",
-    "@inrupt/solid-client-notifications": "*",
-    "next": "12.1.2",
+    "@inrupt/solid-client-notifications": "file:../../../",
```
Comment (ThisIsMissEm) [inline e2e/browser/testApp/package.json:13]: @NSeydoux I switched this to use `file:` instead, after the discussion with the npm team here: https://github.com/npm/feedback/discussions/667

**Pull request merged**